code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -1,512 +1,727 @@
 package minersc
 
 import (
-	"context"
+	"0chain.net/chaincore/smartcontract"
+	"0chain.net/chaincore/state"
+	"0chain.net/smartcontract/stakepool/spenum"
 	"errors"
 	"fmt"
-	"net/url"
+	"net/http"
 	"strconv"
 
-	"0chain.net/chaincore/state"
-
-	"0chain.net/smartcontract/stakepool/spenum"
+	"0chain.net/rest/restinterface"
 
 	"0chain.net/smartcontract/dbs/event"
+	"github.com/guregu/null"
 
 	"0chain.net/core/common"
 	"0chain.net/core/util"
-	"0chain.net/smartcontract"
-	"github.com/guregu/null"
-
-	cstate "0chain.net/chaincore/chain/state"
-
-	. "0chain.net/core/logging"
-	"go.uber.org/zap"
+	sc "0chain.net/smartcontract"
 )
 
+type RestFunctionName int
+
 const (
-	cantGetMinerNodeMsg = "can't get miner node"
+	rfnGlobalSettings RestFunctionName = iota
+	rfnGetNodepool
+	rfnGetUserPools
+	rfnGetMinerList
+	rfnGetMinersStats
+
+	rfnGetMinersStake
+	rfnGetSharderList
+	rfnGetShardersStats
+	rfnGetShardersStake
+	rfnGetSharderKeepList
+
+	rfnGetPhase
+	rfnGetDkgList
+	rfnGetMpksList
+	rfnGetGroupShareOrSigns
+	rfnGetMagicBlock
+
+	rfnGetEvents
+	rfnNodeStat
+	rfnNodePoolStat
+	rfnConfigs
+	rfnGetMinerGeolocations
```

Comment:
The above const variables could get wrong easily if anyone change it's sequence but didn't change the GetRestNames() function. It would also make the upgrade incompatible if we changed switched any positions. So instead of of having this const variables, we should give a list of all rest endpoints with struct that contains the url and corresponding handler, in the `SetupRestHandler()` function, we can just go through the list, and setup all. For the `GetRestNames()` function if needed, we can get the urls from the list and return.

Reply:
Yes, that sounds better.